File: src/main/java/sawyern/cookiebot/commands/RegisterCommand.java
package sawyern.cookiebot.commands;

import discord4j.core.event.domain.message.MessageCreateEvent;
import discord4j.core.object.entity.Member;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import sawyern.cookiebot.models.dto.AccountDto;
import sawyern.cookiebot.exception.CookieException;
import sawyern.cookiebot.services.AccountService;
import sawyern.cookiebot.services.CookieService;

import java.util.List;

@Component
@RequiredArgsConstructor
public class RegisterCommand extends MessageCreateEventBotCommand {

    private final AccountService accountService;
    private final CookieService cookieService;

    @Override
    public void execute(MessageCreateEvent event, List<String> args) throws CookieException {
        Member member = botUtil.getMember(event);
        AccountDto accountDto = AccountDto.builder()
                .discordId(member.getId().asString())
                .username(member.getUsername())
                .build();
        accountService.registerAccount(accountDto);

        for (int i = 0; i < 10; i++)
            cookieService.generateCookie(accountDto.getDiscordId());
        botUtil.sendMessage(event, "Successfully registered " + accountDto.getUsername());
    }

    @Override
    public String getCommand() {
        return "register";
    }
}

File: src/main/kotlin/sawyern/cookiebot/commands/RegisterCommand.kt
package sawyern.cookiebot.commands

import discord4j.core.event.domain.message.MessageCreateEvent
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import sawyern.cookiebot.constants.CommandConstants
import sawyern.cookiebot.constants.Constants
import sawyern.cookiebot.services.AccountService
import sawyern.cookiebot.services.BotUtilService
import sawyern.cookiebot.services.CookieService

@Component
class RegisterCommand @Autowired constructor(
        botUtilService: BotUtilService,
        private val accountService: AccountService,
        private val cookieService: CookieService
): MessageCreateEventBotCommand(botUtilService) {
    override fun getCommand(): String {
        return CommandConstants.CommandName.REGISTER
    }

    override fun getHelpText(): String {
        return "register new account"
    }

    override fun execute(event: MessageCreateEvent, args: List<String>) {
        val member = botUtilService.getMember(event)

        accountService.registerAccount(member.id.asString(), member.username)
        cookieService.generateCookie(member.id.asString(), Constants.Cookie.STARTING_COOKIES)

        botUtilService.sendMessage(event, "Successfully registered " + member.username)
    }
}